compiler/sem/sempass2.nim

```diff
@@ -1621,29 +1621,27 @@ proc detectCapture(owner, top: PSym, n: PNode, marker: var IntSet): PNode =
         # unwind
         return
 
-proc isCompileTimeProc2*(s: PSym): bool =
-  ## Tests if `s` is a compile-time procedure, but compared to
-  ## ``isCompileTimeProc``, also considers where `s` is defined. In the
-  ## following example:
-  ##
-  ## .. code-block:: nim
-  ##
-  ##   proc a() {.compileTime.} =
-  ##     proc b() =
-  ##       discard
-  ##
-  ## ``b`` is also a procedure that can only be run at compile-time, but it
-  ## doesn't have the ``sfCompileTime`` flag set
-  # TODO: propagate the ``sfCompileTime`` flag downwards instead (likely in
-  #       sempass2) and use ``isCompileTimeProc`` directly
-  # FIXME: procedures defined inside ``static`` blocks are not detected
-  var s = s
-  while s.kind != skModule:
-    if isCompileTimeProc(s):
-      return true
+proc canCaptureFrom*(source, dest: PSym): bool =
```

Comment:
```suggestion
proc canCaptureFrom*(captor, trgt: PSym): bool =
```

Suggesting this because I kept finding myself reconfirming if `source` was the routine trying to capture or the source of potential locals.